src/yaml/from_yaml.cr

```diff
@@ -22,39 +29,110 @@ end
 
 def Nil.new(pull : YAML::PullParser)
   location = pull.location
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_plain_scalar
+  when "", "~", "null", "Null", "NULL"
     nil
   else
-    raise YAML::ParseException.new("Expected nil, not #{value}", *location)
+    raise YAML::ParseException.new("Expected null, not '#{value}'", *location)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  location = pull.location
+  case value = pull.read_plain_scalar
+  when "y", "Y", "yes", "Yes", "YES", "true", "True", "TRUE", "on", "On", "ON"
+    true
+  when "n", "N", "no", "No", "NO", "false", "False", "FALSE", "off", "Off", "OFF"
+    false
+  else
+    raise YAML::ParseException.new("Expected boolean, not '#{value}'", *location)
+  end
 end
 
-{% for type in %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) %}
-  def {{type.id}}.new(pull : YAML::PullParser)
+{% for size in %w(8 16 32 64) %}
+  def Int{{size.id}}.new(pull : YAML::PullParser)
     location = pull.location
+    value = pull.read_plain_scalar
     begin
-      {{type.id}}.new(pull.read_scalar)
+      value.to_i{{size.id}}(prefix: true, underscore: true, strict: true)
+    rescue ex
+      raise YAML::ParseException.new(ex.message.not_nil!, *location)
+    end
+  end
+
+  def UInt{{size.id}}.new(pull : YAML::PullParser)
+    location = pull.location
+    value = pull.read_plain_scalar
+    begin
+      value.to_u{{size.id}}(prefix: true, underscore: true, strict: true)
     rescue ex
       raise YAML::ParseException.new(ex.message.not_nil!, *location)
     end
   end
 {% end %}
 
-def String.new(pull : YAML::PullParser)
-  pull.read_scalar
-end
+{% for size in %w(32 64) %}
+  def Float{{size.id}}.new(pull : YAML::PullParser)
+    location = pull.location
+    begin
+      case value = pull.read_plain_scalar
+      when ".inf", ".Inf", ".INF", "+.inf", "+.Inf", "+.INF"
+        INFINITY
+      when "-.inf", "-.Inf", "-.INF"
+        -INFINITY
+      when ".nan", ".NaN", ".NAN"
+        NAN
+      else
+        value.sub('_', "").to_f{{size.id}}
+      end
+    rescue ex
+      raise YAML::ParseException.new(ex.message.not_nil!, *location)
+    end
+  end
+{% end %}
 
-def Float32.new(pull : YAML::PullParser)
-  pull.read_scalar.to_f32
+def Time.new(pull : YAML::PullParser)
+  location = pull.location
+  value = pull.read_plain_scalar
+
+  # Coerce into a format we can parse with Time.parse
```

Comment:
I don't like this. This implementation is partly like a parser but requires a second parser run. You should not try to coerce the value to a specific format but either define a custom parser or leave that for a later PR and stick to `Time::Format::ISO_8601_DATE_TIME` for now.